System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -237,6 +237,10 @@ public override string[] GetFiles(string path, string searchPattern, SearchOptio
             {
                 fileNamePattern = isUnix ? @"[^/]*?/?" : @"[^\\]*?\\?";
             }
+            else if (searchPattern == "*.")
```

Comment:
By checking for `*.`, `MockDirectory` still behaves differently than `System.Directory`. The system version interprets search filters ending in a period as a search for files that either lack an extension or that literally end with a period. Because of this, there are quite a few search filters that won't be fixed by this PR.

For example, `hello*.` should match any filename beginning with `hello` that doesn't have an extension (or ends with just a period). `h????*.` would be similar but would require the filename to be at least a 5 characters long. An even more complex example could be something like `h?*-w*.` which would match a filename like 'hello-world', 'hello-world.' or `hi there-world`.

Reply:
@rpendleton
Thanks for reviewing!
I will think about the examples you provided.
Thanks!

Reply:
Hi, I checked in a new version, which is much closer to how System.Directory behaves.